vendor/github.com/rancher/kontainer-engine/cluster/cluster.go

```diff
@@ -305,7 +305,9 @@ func (c *Cluster) Remove(ctx context.Context) error {
 	}
 
 	if err := c.Driver.Remove(ctx, toInfo(c)); err != nil {
-		return err
+		// Persist store removal must take place despite error to prevent cluster from
+		// being stuck in remove state in UI
```

Comment:
Drop "in UI"
Add "TODO: We should add a "forceRemove" action to clusters and then revert this to returning an error, so that the user can see the problem and take appropriate action?